### PR TITLE
Gris 36.experiment adding author publication

### PR DESCRIPTION
**Pull request by paulinaborys**

### CODE REVIEW

File: grisera_api/experiment/experiment_model.py [hunk 1]
```diff
@@ -10,10 +13,15 @@ class ExperimentIn(BaseModel):
...
         name (str): Name of experiment
+        authors (Optional[Set[AuthorIn]]): Authors of the experiment
+        publication(Optional[PublicationIn]): Publication, in which the experiment is described
         abstract (Optional[str]): Summary of the experiment
         additional_properties (Optional[List[PropertyIn]]): Additional properties for experiment
     """
     name: str
+    authors: Optional[List[AuthorIn]]
+    publication: Optional[PublicationIn]
+    # scenario = Optional[List[Activity]]
```
Comment (mrwrob) [inline grisera_api/experiment/experiment_model.py:24]: Why is it commented? Are we gonna use this scenario later?

File: grisera_api/experiment/experiment_service.py [hunk 2]
```diff
@@ -21,14 +27,38 @@ def save_experiment(self, experiment: ExperimentIn):
...
+            for author in experiment.authors:
+                node_response_author = self.author_service.save_author(self, author=author)
+                # Create relationship between Author and Experiment
+                relationship_response_experiment_author = self.graph_api_service.create_relationships(
+                    end_node=node_response_author.id, start_node=experiment_id, name="hasAuthor")
+                if relationship_response_experiment_author["errors"] is not None:
+                    return ExperimentOut(title=experiment.title, authors=experiment.authors,
+                                         errors=node_response_experiment["errors"])
+
+        if experiment.authors is not None:
```
Comment (mrwrob) [inline grisera_api/experiment/experiment_service.py:51]: Didn't you mean `experiment.publication`?

File: grisera_api/graph_api_service.py [hunk 3]
```diff
@@ -53,10 +53,37 @@ def create_properties(self, node_id: int, node_model: BaseModel):
         """
         node_dict = node_model.dict()
         request_body = [{"key": key, "value": value} for key, value in node_dict.items()
-                        if value is not None and key != 'additional_properties']
+                        if value is not None and key != 'additional_properties' and key != 'authors' and key != 'publication']
```
Comment (mrwrob) [inline grisera_api/graph_api_service.py:56]: I think `.. and key not in ['additional_properties', 'authors', 'publication']` would be more clear

File: grisera_api/graph_api_service.py [hunk 4]
```diff
@@ -53,10 +53,37 @@ def create_properties(self, node_id: int, node_model: BaseModel):
...
+
+        if 'authors' in node_dict and node_dict['authors'] is not None:
+            for authors in node_dict['authors']:
+                request_body.append({"key": "name", "value": authors['name']})
+                request_body.append({"key": "institution", "value": authors['institution']})
+
+        if 'publication' in node_dict and node_dict['publication'] is not None:
+            publication = node_dict['publication']
+            request_body.append({"key": "title", "value": publication['title']})
+            for authors in node_dict['authors']:
```
Comment (mrwrob) [inline grisera_api/graph_api_service.py:70]: I'm not sure if I follow... Doesn't this mean that info about authors will be added twice?

Comment (paulinaborys) [reply grisera_api/graph_api_service.py:70]: I thought that there can be different authors for the publication and for the experiment. That is why the info is added for experiment's authors and for publication's authors.

File: grisera_api/experiment/experiment_model.py (hunk 1 above)
Comment (paulinaborys) [reply grisera_api/experiment/experiment_model.py]: I wasn't sure what should be included as properties in this task. I have just noticed that I didn't also include frequency property. :(

File: grisera_api/graph_api_service.py (hunk 4 above)
Comment (mrwrob) [reply grisera_api/graph_api_service.py:70]: Got it :smile:

**Review by mrwrob — APPROVED** [review]
:clap:

**Pull request merged**